crmd/control.c

```diff
@@ -1061,6 +1062,18 @@ config_query_callback(xmlNode * msg, int call_id, int rc, xmlNode * output, void
         fsa_cluster_name = strdup(value);
     }
 
+    {
+        sub_call_id = fsa_cib_conn->cmds->query(fsa_cib_conn, 
+            "/" XML_TAG_CIB "/" XML_CIB_TAG_CONFIGURATION
+            "/" XML_CIB_TAG_NOTIFICATIONS "/" XML_CIB_TAG_NOTIFY, NULL,
+            cib_scope_local | cib_xpath);
+
+        fsa_register_cib_callback(sub_call_id, FALSE, NULL,
+                                  notifications_query_callback);
+
+        crm_trace("Querying the CIB for notifications ... call %d", sub_call_id);
+    }
```

Comment:
If it's possible, I'd rather expand the crm_read_options() query to search for both crmconfig and alerts in one xpath, and process the results all at once. I'd rather avoid any possible race conditions in two separate queries, and it's nice to save the overhead of two IPC calls.